.github/workflows/ci-pr.yml

```diff
@@ -8,45 +8,34 @@ concurrency:
 jobs:
   ci-pr:
     name: CI-PR
-    needs: [test-minimal, test-full, msrv]
+    needs: [test]
     runs-on: ubuntu-latest
     steps:
       - name: Done
         run: exit 0
-  test-minimal:
-    name: Tests (Minimal)
-    env:
-      FLAGS: --no-default-features --features 'std cargo'
-    strategy:
-      fail-fast: false
-      matrix:
-        os: [ubuntu-latest, windows-latest]
-        rust: [stable]
-    runs-on: ${{ matrix.os }}
+  args:
+    name: Args
+    runs-on: ubuntu-latest
+    outputs:
+      minimal: ${{ steps.minimal.outputs.list }}
+      full: ${{ steps.full.outputs.list }}
     steps:
-      - name: Install rust
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: ${{ matrix.rust }}
-          override: true
-      - name: Cache Builds
-        uses: Swatinem/rust-cache@v1
       - name: Checkout
         uses: actions/checkout@v2
-      - name: Compile
-        run: cargo test --no-run ${{ env.FLAGS }}
-      - name: Test
-        run: cargo test ${{ env.FLAGS }}
-  test-full:
-    name: Tests (Full)
-    env:
-      FLAGS: --features 'wrap_help yaml regex unstable-replace unstable-multicall unstable-grouped'
+      - id: minimal
+        name: Get args (Minimal)
+        run: echo "::set-output name=list::$(make features-minimal)"
+      - id: full
+        name: Get args (Full)
+        run: echo "::set-output name=list::$(make features-full)"
```

Comment:
While I'm pleased with how past the job is (3s), I wish we had visualization to help determine the overall impact of these logistical jobs.

In weighing this proposed implementation complexity with #2901, I feel like the "warnings" isn't really buying us much, but is in fact is harmful (at best clogging up the screen with information not relevant to the current developer, at worst, the current developer things they are responsible and wastes time trying to figure out what to do).

Reply:
There were a few extra PRs external contributors make after seeing these warnings on their screen in their PR. I would say these warnings have been net positive over the last year compared to them being shown at the end of the status check log. They are actually quite helpful sometimes when there's an error because the contributor doesn't have to open the log and scroll. The errors are visible at a glance.

Reply:
Why are we allowing these warnings instead of turning them into errors, forcing them to be resolved in the relevant PR?

Reply:
We do turn a few of the warnings into errors but there are others which are not really that impactful but the end users still see when using our library. We need to maintain a balance here.

Reply:
Balance between what?

Reply:
Balance between what warnings we uplevel as errors and what warnings to leave as such.

Reply:
- If its a warning `clap_derive` users will see by default, we should error on it a lot of people just turn warnings into errors in CI
- Providing non-actionable feedback to a user is confusing and discouraging.  Its helpful to display the warning to the person fixing introducing it but future contributors should not see it.  I'd go as far as saying that if we don't value it enough to turn it into an error, I give more weight to the follow up contributor's experience.

Reply:
Agreed. This is just punting the can down the line. IMO a code contributor should be addressing lint errors in the first place.

Using warnings while compiling code locally during development makes sense because you shouldn't have to be expected to write perfect code while hacking away on something. But if you're asking someone to merge your PR into a project, it makes more sense to address and/or suppress those warnings as needed.

As core maintainers you should expect community members's contributions to meet a certain bar of code quality - marking PRs as failed on a code lint warning is one way to help ensure that.